components/apimgt/org.wso2.carbon.apimgt.spec.parser/src/main/java/org/wso2/carbon/apimgt/spec/parser/definitions/asyncapi/models/AsyncApiV2Parser.java

```diff
@@ -0,0 +1,153 @@
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.apimgt.api.APIDefinitionValidationResponse;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.api.ExceptionCodes;
+import org.wso2.carbon.apimgt.api.model.API;
+import org.wso2.carbon.apimgt.api.model.Scope;
+import org.wso2.carbon.apimgt.spec.parser.definitions.APISpecParserConstants;
+import org.wso2.carbon.apimgt.spec.parser.definitions.AsyncApiParserUtil;
+import org.wso2.carbon.apimgt.spec.parser.definitions.BaseAsyncApiV2Parser;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * New AsyncAPI v2 parser used to process AsyncAPI 2.x specifications.
+ * This class extends BaseAsyncApiV2Parser and
+ * provides version-specific  parsing capabilities for AsyncAPI 2.x definitions.
+ */
+public class AsyncApiV2Parser extends BaseAsyncApiV2Parser {
+
+    private static final Log log = LogFactory.getLog(AsyncApiV2Parser.class);
+
+    /**
+     * Extracts and returns the set of scopes defined in the given AsyncAPI configuration.
+     *
+     * @param resourceConfigsJSON the AsyncAPI configuration in JSON format
+     * @return a set of Scope bjects defined in the configuration
+     * @throws APIManagementException if an error occurs while extracting the scopes
+     */
+    @Override
+    public Set<Scope> getScopes(String resourceConfigsJSON) throws APIManagementException {
+        Set<Scope> scopeSet = AsyncApiParserUtil.getScopesFromAsyncAPIConfig(resourceConfigsJSON);
+        return scopeSet;
+    }
+
+    /**
+     * Validates the given AsyncAPI definition against the AsyncAPI JSON HyperSchema.
+     *
+     * @param apiDefinition      the AsyncAPI definition to be validated, in JSON format
+     * @param returnJsonContent  whether the validated content should be returned in JSON format
+     * @return an APIDefinitionValidationResponse containing validation results and messages
+     * @throws APIManagementException if an error occurs during validation or schema processing
+     */
+    @Override
+    public APIDefinitionValidationResponse validateAPIDefinition(String apiDefinition, boolean returnJsonContent)
+            throws APIManagementException {
+
+        APIDefinitionValidationResponse validationResponse = new APIDefinitionValidationResponse();
+        String protocol = StringUtils.EMPTY;
+        boolean validationSuccess = false;
+        List<String> validationErrorMessages = new ArrayList<>();
+
+        try {
+            validationSuccess = AsyncApiParserUtil.validateAsyncApiContent(apiDefinition, validationErrorMessages);
+        } catch (Exception e) {
+            // unexpected problems during validation/parsing
+            String msg = "Error occurred while validating AsyncAPI definition: " + e.getMessage();
+            log.error(msg, e);
```

Comment:
dont we log this message in calling clases or from REST API layer its self. If so we dont need to explicitly log it again.